tools/packaging/packaging.go

```diff
@@ -540,7 +540,7 @@ func grpcServerForHostname(hostname string) string {
 		return "localhost:8082"
 	case "master.cloud.kolide.net":
 		return "master-grpc.cloud.kolide.net:443"
-	case "kolide.co", "kolide.com":
+	case "kolide.co", "kolide.com", "launcher.kolide.com:443":
```

Comment:
this isn't necessary because it should be satisfied by the default case immediately below:

```
default:
  if strings.Contains(hostname, ":") {
    return hostname
```

Reply:
I beg to differ. I built a package with `--hostname=launcher.kolide.com:443` because I'm used to using the direct hostname, but that created a package where the folder structure was completely different from the packages built with `--hostname=kolide.com`, which resulted in some confusion at first.

Reply:
It works for me as is:

```
$ ./build/package-builder make --hostname=launcher.kolide.com:443
{"caller":"level.go:63","deb":"/tmp/packages_389842957/launcher-linux-0.3.0-10-g58e6807.deb","level":"info","mac":"/tmp/packaging_364122595/launcher-darwin-0.3.0-10-g58e6807.pkg","msg":"created packages","rpm":"/tmp/packages_389842957/launcher-linux-0.3.0-10-g58e6807.rpm","ts":"2017-11-02T17:22:06.917034689Z"}
$ ./tools/packaging/extract_mac_pkg.sh /tmp/packaging_364122595/launcher-darwin-0.3.0-10-g58e6807.pkg
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/etc
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/etc/launcher
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/etc/launcher/secret
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/Library
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/Library/LaunchDaemons
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/Library/LaunchDaemons/com.launcher.launcher.plist
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/usr
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/usr/local
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/usr/local/launcher
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/usr/local/launcher/bin
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/usr/local/launcher/bin/launcher
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/usr/local/launcher/bin/osquery-extension.ext
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/usr/local/launcher/bin/osqueryd
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/var
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/var/launcher
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/var/launcher/launcher.kolide.com-443
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/var/log
/var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/var/log/launcher
$ cat /var/folders/wp/6fkmvjf11gv18tdprv4g2mk40000gn/T/tmp.OBWitWAw/package/Library/LaunchDaemons/com.launcher.launcher.plist
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
    <dict>
        <key>Label</key>
        <string>com.launcher.launcher</string>
        <key>EnvironmentVariables</key>
        <dict>
            <key>KOLIDE_LAUNCHER_ROOT_DIRECTORY</key>
            <string>/var/launcher/launcher.kolide.com-443</string>
            <key>KOLIDE_LAUNCHER_HOSTNAME</key>
            <string>launcher.kolide.com:443</string>
            <key>KOLIDE_LAUNCHER_ENROLL_SECRET_PATH</key>
            <string>/etc/launcher/secret</string>
            <key>KOLIDE_LAUNCHER_OSQUERYD_PATH</key>
            <string>/usr/local/launcher/bin/osqueryd</string>
        </dict>
        <key>KeepAlive</key>
        <dict>
            <key>PathState</key>
            <dict>
                <key>/etc/launcher/secret</key>
                <true/>
            </dict>
        </dict>
        <key>ThrottleInterval</key>
        <integer>60</integer>
        <key>ProgramArguments</key>
        <array>
            <string>/usr/local/launcher/bin/launcher</string>



        </array>
        <key>StandardErrorPath</key>
        <string>/var/log/launcher/launcher-stderr.log</string>
        <key>StandardOutPath</key>
        <string>/var/log/launcher/launcher-stdout.log</string>
    </dict>
</plist>%                                                                                                                                                                                                                                                                       $
```

Reply:
That's not the correct rundir/hostname directory.